ckan/config/middleware/flask_app.py

```diff
@@ -24,6 +58,81 @@ def make_flask_stack(conf, **app_conf):
         app.config.update(conf)
         app.config.update(app_conf)
 
+    # Do all the Flask-specific stuff before adding other middlewares
+
+    # Secret key needed for flask-debug-toolbar and sessions
+    if not app.config.get('SECRET_KEY'):
+        app.config['SECRET_KEY'] = config.get('beaker.session.secret')
+    if not app.config.get('SECRET_KEY'):
+        raise RuntimeError(u'You must provide a value for the secret key'
+                           ' with the SECRET_KEY config option')
+
+    if debug:
+        app.config['DEBUG_TB_INTERCEPT_REDIRECTS'] = False
+        DebugToolbarExtension(app)
+
+    # Add Jinja2 extensions and filters
+    extensions = [
+        'jinja2.ext.do', 'jinja2.ext.with_',
+        jinja_extensions.SnippetExtension,
+        jinja_extensions.CkanExtend,
+        jinja_extensions.CkanInternationalizationExtension,
+        jinja_extensions.LinkForExtension,
+        jinja_extensions.ResourceExtension,
+        jinja_extensions.UrlForStaticExtension,
+        jinja_extensions.UrlForExtension
+    ]
+    for extension in extensions:
+        app.jinja_env.add_extension(extension)
+    app.jinja_env.filters['empty_and_escape'] = \
+        jinja_extensions.empty_and_escape
+    app.jinja_env.filters['truncate'] = jinja_extensions.truncate
+
+    # Common handlers for all requests
+    @app.before_request
+    def ckan_before_request():
+        # Start the request timer
+        g._request_timer = time.time()
+
+        # Update app_globals
+        app_globals.app_globals._check_uptodate()
+
+        # Identify the user from the repoze cookie or the API header
+        # Sets g.user and g.userobj
+        identify_user()
+
+    @app.after_request
+    def ckan_after_request(response):
+        # Check session cookie
+        response = check_session_cookie(response)
+
+        # Set CORS headers if necessary
+        response = set_cors_headers_for_response(response)
+
+        # Log time between before and after view
```

Comment:
Not a fan of this being a built-in/default behaviour. If you're developing/debugging it's part of Flask-debugtoolbar already and if you're in production it's part of whatever layer you're using above CKAN or 99% of tools like Sentry for logging. Would just get rid of this.